doc/explanation/components/reactive_html_components.md

```diff
@@ -1,12 +1,260 @@
+import panel as pn
+pn.extension() # for notebook
+
+class Slideshow(ReactiveHTML):
+
+    index = param.Integer(default=0)
+
+    _template = '<img id="slideshow" src="https://picsum.photos/800/300?image=${index}" onclick="${_img_click}"></img>'
+
+    def _img_click(self, event):
+        self.index += 1
+
+Slideshow(width=800, height=300)
+```
+
+## Where do I find practical examples?
+
+To see `ReactiveHTML` in action and discover **how to** create your custom components, check out our detailed guide: [How-to > Create Custom Components with ReactiveHTML](../../how_to/custom_components/reactive_html/index.md). It's packed with practical examples to help you get started quickly.
+
+## Where do I find the full API description?
+
+You can find it here [API > ReactiveHTML](../../api/panel.reactive.html#panel.reactive.ReactiveHTML).
+
+## What class attributes should a component declare?
+
+The component must declare
+
+- `_template`: The *blueprint* for how your custom component should look and behave. The *_template* is defined using HTML. You can use [Jinja2](https://jinja.palletsprojects.com) syntax as well as *template variables* `${...}` to make the template *dynamic*.
+
+The component may optionally declare
+
+- `_child_config` (dict): Mapping that controls how children are rendered.
+- `_dom_events` (dict): Mapping of node ids to DOM events to add event listeners to.
+- `_extension_name` (str): Name used to import external `css` and `js` dependencies via `pn.extension(_extension_name)`.
+- `_scripts` (dict): Mapping of javascript scripts. These scripts are automatically executed during the component's life cycle and on parameter changes.
+
+The component may also optionally declare
+
+- `__css__`: A list of CSS dependencies required to style your component.
+- `__javascript__`: A list of javascript dependencies that your component relies on.
+- `__javascript_modules__`: A list of javascript module dependencies that your component relies on.
+
+### What does the `_template` attribute do?
+
+The `_template` variable defines the *blueprint* of your component using HTML (content) and optionally CSS (style) or Javascript (behaviour).
+
+```python
+  _template = '<img id="slideshow" src="https://picsum.photos/800/300?image=${index}" onclick="${_img_click}"></img>'
+```
+
+The `_template` supports Python [Jinja2](https://jinja.palletsprojects.com/en/2.11.x/) syntax as
+well as Javascript *template variables* `${...}`.
+
+#### What can I use Jinja2 templating for?
+
+You can use Jinja2 syntax to layout your template. When using Jinja2 syntax you can refer to parameters using `{{ ... }}` syntax. This will insert your parameter values as a literal string values. In addition you can use the following context variables:
+
+- `param`: The param namespace object allows templating parameter names, labels, docstrings and other attributes.
+- `__doc__`: The class docstring
+
+Here is an illustrative example
+
+```{pyodide}
```

Comment:
This example is important. I spent hours finding some combination that works.

One problem is that `ReactiveHTML` has issues inserting `{{ ... }}` values that contains tilde

- https://github.com/holoviz/panel/issues/5499
- https://github.com/holoviz/panel/issues/5500

Another is that I could not find a working way to just loop `param`. Instead I had to use `param.params().values()` which is cumbersome.

Another is that I could not find an easy way to get the current value of a parameter in the loop. So I ended up having to do `object.owner[object.name]`.

Reply:
> One problem is that ReactiveHTML has issues inserting {{ ... }} values that contains tilde

Will look at fixing that.

> Another is that I could not find a working way to just loop param. Instead I had to use param.params().values() which is cumbersome.

`.param` can be looped directly.

> Another is that I could not find an easy way to get the current value of a parameter in the loop. So I ended up having to do object.owner[object.name].

This is a good point, `.param.values()[name]` seems awkward.

Reply:
You can loop `.param` directly. It gives you a name of a parameter. But `.param[name]` does not seem to work. Thus you cannot get the actual parameter to work with.